app/goal_routes.py

```diff
@@ -0,0 +1,126 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+import requests
+
+
+
+goal_bp = Blueprint("goal_bp", __name__, url_prefix='/goals')
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():        
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response(jsonify({"details":"Invalid data"})),400
+    new_goal= Goal(
+        title = request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }},201
+
+@goal_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals = Goal.query.all()
+    
+    goals_response = []
+    for goal in goals:
+        goals_response.append({
+            "id": goal.goal_id,
+            "title": goal.title
+            })
```

Comment:
I see that this code _works_ (and I see why it _works_) but it requires focused study to understand it. Could you please refactor it to use list comprehension so it will be easier to maintain in the future?